shared/js/views.jsx

```diff
@@ -599,22 +599,94 @@ loop.shared.views = (function(_, mozL10n) {
       isLoading: React.PropTypes.bool.isRequired,
       mediaType: React.PropTypes.string.isRequired,
       posterUrl: React.PropTypes.string,
+      remoteCursorLeft: React.PropTypes.number,
+      remoteCursorTop: React.PropTypes.number,
       // Expecting "local" or "remote".
       srcMediaElement: React.PropTypes.object
     },
 
+    getInitialState: function() {
+      return {
+        offsetLeft: 0,
+        offsetTop: 0,
+        streamVideoHeight: 0,
+        streamVideoWidth: 0
+      };
+    },
+
+    componentWillReceiveProps: function(nextProps) {
+      if (!nextProps.remoteCursorLeft || !nextProps.remoteCursorTop) {
+        return;
+      }
+
+      // We need to calculate the cursor postion based on the current video stream dimensions
+      var percentX = nextProps.remoteCursorLeft;
+      var percentY = nextProps.remoteCursorTop;
+
+      var cursorPositionX = (this.state.streamVideoWidth * percentX) / 100;
+      var cursorPositionY = (this.state.streamVideoHeight * percentY) / 100;
+console.info("cursor position X", cursorPositionX);
+console.info("cursor position Y", cursorPositionY);
+      this.setState({
+        cursorPositionX: cursorPositionX,
+        cursorPositionY: cursorPositionY
+      });
+    },
+
     componentDidMount: function() {
       if (!this.props.displayAvatar) {
         this.attachVideo(this.props.srcMediaElement);
       }
     },
 
+    componentWillUnmount: function() {
+      var videoElement = this.getDOMNode().querySelector("video");
+      videoElement.removeEventListener("timeupdate", this.handleVideoUpdate);
```

Comment:
We add the listener in |attachVideo| and this method it's just to cleanup the listeners before react remove the element from the dom. Perhaps we can rid off it, wdyt?